census/census.go

```diff
@@ -65,15 +76,30 @@ type Manager struct {
 	failedQueueLock sync.RWMutex
 	failedQueue     map[string]string
 	compressor
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
 }
 
 // Data helps satisfy an ethevents interface.
 func (m *Manager) Data() data.Storage { return m.RemoteStorage }
 
+// BadgerDBBatched wraps badgerdb.BadgerDB turning all requested db.WriteTx
+// into db.Batch so that commits happen automatically when the Tx becomes too
+// big.
+type BadgerDBBatched struct {
+	*badgerdb.BadgerDB
+}
+
+// WriteTx returns a db.Batch that will commit automatically with the Tx
+// becomes too big.
+func (d *BadgerDBBatched) WriteTx() db.WriteTx {
+	return db.NewBatch(d.BadgerDB)
+}
+
 // Init creates a new census manager.
 // A constructor function for the interface censustree.Tree must be provided.
-func (m *Manager) Init(storageDir, rootAuthPubKey string) error {
-	nsConfig := fmt.Sprintf("%s/namespaces.json", storageDir)
+func (m *Manager) Init(storageDir, rootAuthPubKey string) (err error) {
+	nsConfig := filepath.Clean(filepath.Join(storageDir, "namespaces.json"))
```

Comment:
not sure why we have all these `filepath.Clean` invocations. if we're worried that `storageDir` has bits that cancel out, like `foo/bar/..`, we should clean those at startup and avoid calling Clean again in five separate places.

Reply:
Makes sense.  I didn't remove it because I though it was there for some reason, I'll see if I can move it upwards.

Reply:
I realise some of what I bring up is not your own doing. But since we're here... :)